### PR TITLE
[spark] check column nullability when write

### PR DESCRIPTION
**Pull request by YannByron**
### Purpose

Linked issue: close #xxx

### CODE REVIEW

File: paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/SparkConnectorOptions.java [hunk 1]
```diff
@@ -39,6 +39,13 @@ public class SparkConnectorOptions {
                     .withDescription(
                             "If true, allow to merge data types if the two types meet the rules for explicit casting.");
 
+    public static final ConfigOption<Boolean> IGNORE_NULLABLE_CHECK =
+            key("write.check.ignoreNullability")
+                    .booleanType()
+                    .defaultValue(false)
```
Comment (JingsongLi) [inline paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/SparkConnectorOptions.java:45]: Why not just be true?

Comment (YannByron) [reply paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/SparkConnectorOptions.java:45]: I want to have this check by default to guarantee the nullability consistency.

And, Delta has `constraint` feature (eg: NOT NULL) which can check the incoming data when write. When paimon has this similar feature, we can remove this config and do not check nullability any more.

Comment (JingsongLi) [reply paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/SparkConnectorOptions.java:45]: Does this option affect compilation? I thinking about checking for null directly at runtime, but checking at compile time is too difficult to use.

**Review by JingsongLi — APPROVED** [review]
Looks good to me!

**Pull request merged**